infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -503,7 +546,7 @@ private void onLeadershipAcquire() {
             case STANDBY:
                 // Standby Site : the LogReplicationServer (server handler) will initiate the LogReplicationSinkManager
                 log.info("Start as Sink (receiver)");
-                interClusterReplicationService.getLogReplicationServer().getSinkManager().reset();
+                clientToSinkManagerMap.values().forEach(sinkManager -> sinkManager.reset());
```

Comment:
Here is where I say it does not make much sense. Imagine this instance was the active for 3 standby's at T0, so when this node came up you created 3 SinkManager's (one per remote). Then when this becomes the standby, you're 'resetting', overall, operating over the sink manager for each remote. But this became standby to only one of these, the other two instances are useless and incorrect to be operating on.